File: benchmarks/wrappers/fission/python/handler.py
from flask import request, jsonify, current_app

import json, datetime
def handler():
    body = request.get_data().decode("utf-8")
    current_app.logger.info("Body: " + body)
    event = json.loads(body)
    current_app.logger.info("Event: " + str(event))
    begin = datetime.datetime.now()
    from function import function
    ret = function.handler(event)
    end = datetime.datetime.now()
    current_app.logger.info("Function result: " + str(ret))
    log_data = {
        'result': ret['result']
    }
    if 'measurement' in ret:
        log_data['measurement'] = ret['measurement']
    results_time = 0
    # how to check it? 
    is_cold = False
    return jsonify(
        json.dumps({
            'begin': begin.strftime('%s.%f'),
            'end': end.strftime('%s.%f'),
            'results_time': results_time,
            'is_cold': is_cold,
            'result': log_data  
    }))
    

File: benchmarks/wrappers/fission/nodejs/handler.js
const path = require('path'), fs = require('fs');

exports.handler = async function(context) {
    var body = JSON.stringify(context.request.body);
    var unbody = JSON.parse(body);
    var func = require('./function/function');
    var begin = Date.now()/1000;
    var start = process.hrtime();
    var ret = await func.handler(unbody);
    var elapsed = process.hrtime(start);
    var end = Date.now()/1000;
    var micro = elapsed[1] / 1e3 + elapsed[0] * 1e6;
    var is_cold = false;
    var fname = path.join('/tmp','cold_run');
    if(!fs.existsSync(fname)) {
        is_cold = true;
        fs.closeSync(fs.openSync(fname, 'w'));
    }

    return {
        status: 200,
        body: JSON.stringify({
            begin,
            end,
            compute_time: micro,
            results_time: 0,
            result: ret,
            is_cold
        })
    };
};